b2/bucket.py

```diff
@@ -271,6 +273,58 @@ def upload_local_file(
         upload_source = UploadSourceLocalFile(local_path=local_file, content_sha1=sha1_sum)
         return self.upload(upload_source, file_name, content_type, file_infos, progress_listener)
 
+    def upload_stream(
+        self,
+        file_name,
+        content_type=None,
+        file_infos=None
+    ):
+        """
+        Uploads a file from stdin stream to a B2 file.
+        """
+        #determine if it requires large file upload, >= part_size >= minimum_part_size
+        #TODO make part_size user configurable 100MB to 5GB so max file size of 10TB can be reached with 1 to 10000 parts
+        part_size = self.api.account_info.get_minimum_part_size()
+        validate_b2_file_name(file_name)
+        file_info = file_infos or {}
+        content_type = content_type or self.DEFAULT_CONTENT_TYPE
+        
+        data = sys.stdin.read(part_size)
+        #check if small file upload
+        if len(data) < part_size:
+            #TODO need to implement small file upload
+            print 'Starting small file upload'
+        else:
+            print 'Starting large file multipart upload'
+            unfinished_file = self.start_large_file(file_name, content_type, file_info)
+            part_number = 0
+            sha1_array = []
+            while True:
+                upload_url, upload_auth_token = self._get_upload_part_data(unfinished_file.file_id)
+                sha1_sum = hex_sha1_of_bytes(data)
+                sha1_array.append(sha1_sum)
+                part_number += 1
+                print 'Part %d of size %d bytes being uploaded ...' %(part_number, len(data))
+                response = self.api.raw_api.upload_part(
```

Comment:
Can you use the existing Bucket._upload_part() method here?  It already knows how to compute the sha1, and has hooks for progress bars.